learn/getting-started/writing-a-grpc-service-with-ballerina.md

```diff
@@ -15,70 +15,91 @@ redirect_from:
 
 ## Setting up the prerequisites
 
-To complete this tutorial, you need:
+>**Info:** You need a command terminal to complete this tutorial.
 
-1. A command terminal
-2. A text editor
-    >**Tip:** Preferably, [Visual Studio Code](https://code.visualstudio.com/) with the [Ballerina extension](https://marketplace.visualstudio.com/items?itemName=WSO2.ballerina) installed.
-3. A [Ballerina installation](https://ballerina.io/learn/installing-ballerina/setting-up-ballerina/)
+To set up the other prerequisites, follow these steps:
+
+1. Install [Ballerina](https://ballerina.io/learn/installing-ballerina/setting-up-ballerina/).
+2. Install a text editor. 
+  >**Tip:** Preferably, [Visual Studio Code](https://code.visualstudio.com/) with the [Ballerina extension](https://marketplace.visualstudio.com/items?itemName=WSO2.ballerina) installed.
+
+## Understanding the implementation
+
+In an RPC program, you first define the service interface using an Interface Definition Language(IDL) to create the service definition (i.e., `helloworld.proto`). gRPC commonly uses Protocol Buffers as the IDL.
+
+As illustrated in the diagram below, next, you compile the service definition file (i.e., `helloworld.proto`), and generate the source code for both the service and client applications. In Ballerina, you can generate the source code using the built-in `Protocol Buffers to Ballerina` tool.
+
+![gRPC Getting Started](/learn/images/grpc-getting-started.png)
 
 ## Creating the service definition
 
-In an RPC program, the first step is to define the service interface using an Interface Definition Language(IDL).
-gRPC commonly uses Protocol Buffers as the IDL. Let's write a simple service definition in Protocol Buffers as below.
+To create a simple service definition in Protocol Buffers, follow these steps:
 
-```proto
-syntax = "proto3";
+> **Info:** The sample service definition is taken from the [quick start guide on the gRPC official site](https://grpc.io/docs/languages/go/quickstart/).
 
-package helloworld;
+1. Create a new directory named `grpc` in a preferred location (this is your main directory.)
```

Comment:
Shall we use a different name like `grpc_sample` as this name can be a confusing one with `bal grpc` command. As command  name  and folder name are both same.

Reply:
Done.